File: Application/src/main/java/com/example/android/mediasession/service/PlaybackInfoListener.java
package com.example.android.mediasession.service;

import android.support.v4.media.session.PlaybackStateCompat;
import android.support.v4.media.session.MediaSessionCompat;

import com.example.android.mediasession.service.players.MediaPlayerAdapter;

public abstract class PlaybackInfoListener {

    public abstract void onPlaybackStateChange(PlaybackStateCompat state);

    public void onPlaybackCompleted() {
    }
}

File: Application/src/main/java/com/example/android/mediasession/service/PlaybackInfoListener.kt
package com.example.android.mediasession.service

import android.support.v4.media.session.PlaybackStateCompat
import android.support.v4.media.session.MediaSessionCompat

import com.example.android.mediasession.service.players.MediaPlayerAdapter

abstract class PlaybackInfoListener {

    abstract fun onPlaybackStateChange(state: PlaybackStateCompat)

    fun onPlaybackCompleted() {}
}